pca.go

```diff
@@ -0,0 +1,81 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package stat
+
+import (
+	"math"
+
+	"github.com/gonum/floats"
+	"github.com/gonum/matrix"
+	"github.com/gonum/matrix/mat64"
+)
+
+// PrincipalComponents returns the principal component direction vectors and
+// the variances of the principle component scores, vecs * a, computed using
```

Comment:
The diagonal of vecs * a? vecs * a is a d x min(n,d) matrix, while vars is one dimensional.